### PR TITLE
tests: drivers: more testcases running on the nucleo_l152re board

### PR DESCRIPTION
**Pull request by FRASTM**
Add the support of new tests/drivers on the nucleo_l152re target board
the `tests/drivers/spi/spi_loopback` testcase 
the `tests/drivers/uart/uart_async_api` testcase 

The dma1 of type v2bis is added to the dts for the stm32l1 soc serie
The tests/drivers/spi/spi_loopback is configured to run on the nucleo_l152re target. The spi1 MISO pin must be connected to the spi1 MOSI pin to pass the test.
The tests/drivers/uart/uart_async_api is configured to run on the nucleo_l152re target. The usart3 Tx pin must be connected to the Rx pin to pass the test.

Signed-off-by: Francois Ramu francois.ramu@st.com

### CODE REVIEW

**Comment by galak** [comment]
Marked this DNM as not clear if all the commits in here are meant, based on the subject message of adding SPI loopback test support.

**Comment by FRASTM** [comment]
rebased and redefining the PR after the merge of the new dma bindings.
This PR completes the test coverage on another stm32l1 soc serie.

**Review by ABOSTM — APPROVED** [review]
LGTM

File: tests/drivers/spi/spi_loopback/boards/nucleo_l152re.overlay [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+};
+
+&dma1 {
+	status = "okay";
+};
+
+&spi1 {
+	pinctrl-0 = <&spi1_nss_pa4 &spi1_sck_pa5
+		     &spi1_miso_pa6 &spi1_mosi_pa7>;
+	status = "okay";
```
Comment (MaureenHelm) [inline tests/drivers/spi/spi_loopback/boards/nucleo_l152re.overlay:20]: Merge this into the first `spi1` above

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: tests/drivers/spi/spi_loopback/boards/nucleo_l152re.overlay (hunk 1 above)
Comment (FRASTM) [reply tests/drivers/spi/spi_loopback/boards/nucleo_l152re.overlay]: ok

**Review by erwango — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Comment by erwango** [comment]
> commits in here are meant, based on the subject message of adding SPI loopback test support.

@galak Seems the comment has been addressed. Can you consider removing `DNM` ?

**Comment by FRASTM** [comment]
rebase

**Comment by FRASTM** [comment]
not to close

**Review by dcpleung — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Comment by FRASTM** [comment]
- Define the pinning of the usart3 : change in the nucleo_l152re  board overlay for the `uart_async_api` test case:
Tx and Rx pins must be wired (pin 25 and pin 18 of the CN10 connector) to PASS the test.
Note that the dma1 is already enabled in the board DTS.
- Specify the usart3 instance in the `tests/drivers/uart/uart_async_api/src/test_uart.h`
- Use the macro from the include/drivers/dma/dma_stm32.h in 
`.dma_slot = STM32_DMA_SLOT(index, dir, slot),`
Else an compilation error occurs in the uart_stm32.c because, for dma instance of type V2bis (like stm32l152 soc), the dma-cell has NO slot, then it must be defined as null by this macro

**Review by dcpleung — APPROVED** [review]

**Pull request merged**